robottelo/vm.py

```diff
@@ -52,25 +58,34 @@ def __init__(
             image_dir=None):
         self.cpu = cpu
         self.ram = ram
-        if distro is None or distro not in BASE_IMAGES:
-            self.distro = BASE_IMAGES[0]
-        else:
-            self.distro = distro
+        self.distro = BASE_IMAGES[-1] if distro is None else distro
+        if self.distro not in BASE_IMAGES:
+            raise VirtualMachineError(
+                u'{0} is not a supported distro. Choose one of {1}'
+                .format(self.distro, ', '.join(BASE_IMAGES))
+            )
         if provisioning_server is None:
             self.provisioning_server = conf.properties.get(
                 'clients.provisioning_server')
         else:
             self.provisioning_server = provisioning_server
+        if self.provisioning_server is None:
+            raise VirtualMachineError(
+                'A provisioning server must be provided. Make sure to fill '
+                '"provisioning_server" on clients section of your robottelo '
+                'configuration. Or provide a not None provisioning_server '
+                'argument.'
+            )
         if image_dir is None:
             self.image_dir = conf.properties.get(
                 'clients.image_dir', '/var/lib/libvirt/images/')
         else:
             self.image_dir = image_dir
 
-        self.target_image = str(id(self))
-        self.ip_addr = None
         self.hostname = None
+        self.ip_addr = None
         self._created = False
+        self._target_image = str(id(self))
```

Comment:
This change will break existing tests. As ```self.hostname``` was broken before your PR (thanks for fixing it, btw), tests in ```cli.contenthost.TestCHKatelloAgent``` were using ```self.target_image``` instead. 
Please, ```s/self.vm.target_image/self.vm.hostname``` [here](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_contenthost.py#L637) and [here](https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/cli/test_contenthost.py#L664)